File: src/main/java/com/contratos/model/Agreement.java
package com.contratos.model;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;
import java.time.Period;

@EqualsAndHashCode
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
public class Agreement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NonNull
    private Long id;
    @OneToOne
    @NonNull
    private Institute institute;
    @OneToOne
    @NonNull
    private Category category;
    @NonNull
    private Double points;
    @NonNull
    @NotNull
    private AgreementDurationType durationType;
    @NonNull
    private Period duration;
    @NonNull
    private LocalDate initialDate;
    @NonNull
    private LocalDate endDate;
    @NonNull
    private LocalDate assignedDate;
    @NonNull
    @NotNull
    private Boolean accepted;
}

File: src/js/contratos-client/src/app/agreement.ts
import { Institute } from "./institute";
import { Category } from "./category";
import { AgreementDurationType } from "./agreementDurationType";

export class Agreement {
    id: number;
    institute: Institute;
    category: Category;
    points: number;
    duration: String;
    initialDate: String;
    endDate: String;
    assignedDate: String;
    durationType: AgreementDurationType;
    accepted: boolean;

    constructor() {
        this.id = 0;
        this.institute = new Institute();
        this.category = new Category();
        this.points = 0;
        this.duration = '';
        this.initialDate = '';
        this.endDate = '';
        this.assignedDate = '';
        this.durationType = AgreementDurationType.SHORT;
        this.accepted = true;
    }
}

